component/src/main/java/me/androidbox/component/general/PasswordEntry.kt

```diff
@@ -0,0 +1,107 @@
+package me.androidbox.component.general
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import me.androidbox.component.R
+import me.androidbox.component.ui.theme.*
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun PasswordEntry(
+    modifier: Modifier = Modifier,
+    passwordValue: String,
+    placeholderText: String,
+    onPasswordChange: (String) -> Unit,
+    onPasswordVisibilityClicked: () -> Unit,
+    isPasswordVisible: Boolean
+) {
+
+    OutlinedTextField(
+        modifier = modifier,
+        singleLine = true,
+        value = passwordValue,
+        onValueChange = { newInput: String ->
+            onPasswordChange(newInput)
+        },
+        placeholder = {
+            Text(text = placeholderText, color = MaterialTheme.colorScheme.placeholderEntry)
+        },
+        trailingIcon = {
+            val visibilityIconId = if(isPasswordVisible) {
+                R.drawable.visible
+            } else {
+                R.drawable.hidden
+            }
+            Icon(
```

Comment:
When you add Clickable to an Icon, it's recommended to use the composable "IconButton", since it already has the properties adapted to fit the thumb and some extra considerations